neqo-transport/src/send_stream.rs

```diff
@@ -34,7 +34,7 @@ use crate::{
     AppError, Error, Res,
 };
 
-pub const SEND_BUFFER_SIZE: usize = 0x10_0000; // 1 MiB
+const MAX_SEND_BUFFER_SIZE: usize = 10 * 1024 * 1024;
```

Comment:
Previously Neqo would buffer at most 1 MB of send data. Now Neqo buffers up to 10 MB. In other words, it supports an up to 10 MB large send window, depending on the receive window updates of the receiver.

Thus, while this pull request is focused on increasing receive (download) throughput, this patch might as well have an impact on send (upload) throughput on high bandwidth-delay product connections.